### PR TITLE
#12095 Increase RSA keys generated in tests

### PR DESCRIPTION
**Pull request by alex**
This will allow pyca/cryptography to increase the minimum key size. 512-bit keys are wildly insecure, so we want to disallow them.

### CODE REVIEW

**Comment by adiroiban** [comment]
Thanks Alex. Let me know when this is ready for review.

I guess that this is a change needes as in the future cryptography will no longer allow keys below 1k.

Cheers

**Comment by alex** [comment]
This is ready for review now (unless you think it's missing something?)

And yes, this is to keep twisted tests green on a future release -- twisted itself is fine, just the tests are impacted

**Review by adiroiban — APPROVED** [review]
I am happy to have this merged.
Much appreciated for including twisted into the `cryptography` test suite.

-------

Is there a GitHub issue, or discussion related to this change?

I guess that this is a follow up of PR https://github.com/twisted/twisted/pull/1487  ... this updates KEX part.

--------

I know that in the past @glyph has insisted on created a separate/dedicated issue for each change.
But I think that for these code cleanup, it's ok to only have a PR, as long as the PR has a descriptions.

This PR has no description, so I would prefer to merge this only after we add a few words about these changes from here.

thanks again!

PS: Many thanks for the cryptography 42 release :p

**Comment by alex** [comment]
Ooops, there was some extra text in teh commit message that I acciddentally removed from the PR description.

Let me know if that looks good -- if so I'll merge.

**Comment by adiroiban** [comment]
Many thanks. I have merged this.

**Comment by exarkun** [comment]
> I know that in the past @glyph has insisted on created a separate/dedicated issue for each change. But I think that for these code cleanup, it's ok to only have a PR, as long as the PR has a descriptions.

This isn't just glyph's insistence.  This is project policy.  It's part of the responsibility of committers to adhere to project policy.  If you think there should be a variance, take it up on the list to get an exception or have the overall policy changed.

**Comment by adiroiban** [comment]
True. Thanks for the heads up.
I start a discussion to change the policy at https://mail.python.org/archives/list/twisted@python.org/thread/L6GEKQYSWUX4NBIHYEML3AKB2PXDD56N/

**Pull request merged**
